### ISSUE TITLE
[PHP 7.1] Detect nullable return types and warn if passed as argument

### ISSUE THREAD

**Issue by ruudk** [kalessil/phpinspectionsea#77]
For example, I have the following method:

```php
function getText() : ?string
{
    return $this->text;
}
```

When I use this method to store it inside a variable:
```php
$text = $object->getText();
```

And then pass it to another function:
```php
function otherMethod(string $text)
{
    // This method requires a string input
}

otherMethod($text);
```

It would be nice if that would show a warning, when I didn't check for `null` before passing it to the method.

Don't know if this is possible. But if would be super awesome :D

**Comment by funivan** [comment]
I think phpstorm Ide can solve this issue. They already have some type check l

**Comment by kalessil** [comment]
Yep, PS team working on the scope of strict types and I'd suggest creating a ticket in their bug-tracker: https://youtrack.jetbrains.com/issues/WI
